File: algo-c-to-_/examples/moveblock.lua
--
--	from src/movebloc.c
--
--	void reverse(int, int)		to	reverse
--	void rotate(int, int, int)	to	rotate
--

local M = require 'moveblock'

local reverse, rotate = M.reverse, M.rotate

do
	local s = "SUPERCALIFRAGILISTICEXPIALIDOCIOUS"

	local a = {}
	for i=1,#s do a[i] = s:sub(i,i) end

	reverse(a, 1, #a)
	reverse(a, 1, #a)
	assert(s == table.concat(a))

	print(table.concat(a))
	for _=1,17 do
		rotate(a, 1, 6, #a)
		print(table.concat(a))
	end
end

File: algo-c-to-_/examples/moveblock_.py
#
#	from src/movebloc.c
#
#	void reverse(int, int)		to	reverse
#	void rotate(int, int, int)	to	rotate
#

from moveblock import reverse, rotate
from _helper import decrement

#

s = "SUPERCALIFRAGILISTICEXPIALIDOCIOUS"
a = list(s)
n = decrement(len(a))

reverse(a, 0, n)
reverse(a, 0, n)
assert s == "".join(a)

print("".join(a))
for _ in range(17):
    rotate(a, 0, 5, n)
    print("".join(a))
